src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -248,24 +248,34 @@ private bool StartCore(ProcessStartInfo startInfo)
             {
                 Debug.Assert(stdinFd >= 0);
                 _standardInput = new StreamWriter(OpenStream(stdinFd, FileAccess.Write),
-                    new UTF8Encoding(encoderShouldEmitUTF8Identifier: false), StreamBufferSize) { AutoFlush = true };
+                    GetEncoding(), StreamBufferSize) { AutoFlush = true };
             }
             if (startInfo.RedirectStandardOutput)
             {
                 Debug.Assert(stdoutFd >= 0);
                 _standardOutput = new StreamReader(OpenStream(stdoutFd, FileAccess.Read),
-                    startInfo.StandardOutputEncoding ?? Encoding.UTF8, true, StreamBufferSize);
+                    startInfo.StandardOutputEncoding ?? GetEncoding(), true, StreamBufferSize);
             }
             if (startInfo.RedirectStandardError)
             {
                 Debug.Assert(stderrFd >= 0);
                 _standardError = new StreamReader(OpenStream(stderrFd, FileAccess.Read),
-                    startInfo.StandardErrorEncoding ?? Encoding.UTF8, true, StreamBufferSize);
+                    startInfo.StandardErrorEncoding ?? GetEncoding(), true, StreamBufferSize);
             }
 
             return true;
         }
 
+        /// <summary>Creates an encoding from the current environment.</summary>
+        /// <returns>The encoding.</returns>
+        private static Encoding GetEncoding()
+        {
+            Encoding enc = EncodingHelper.GetEncodingFromCharset();
```

Comment:
Hmm.. thought about this a little more and I am not sure we can fully support this on x-plat w/o taking dependency on Console contract.

Process class uses Console encodings (I/p and o/p) to initialize its streams in case PSI does not point to anything. Also, Console encodings are cached unless they are changed using Console encoding setters (i.e we only honor changes done to Console encodings via our managed layer). Currently, we do not expose Console encodings and so, we could cache the value in the Process and always use that. However, we intend to expose Console encoding APIs eventually and since, we do not know a good way to set the console terminal encodings on x-plat, our implementation would most likely only store the encoding in Console class and create console stream using it. Since, this change will be restricted to Console managed layer, there will be no way Process class will know about this encoding w/o querying Console APIs directly. 

So, we have a few options right now
1. Let Process on x-plat continue to use UTF8 (old behavior). This keeps the story simple and if the user wants to change the o/p , error encoding, they have the option of using the PSI.
2. Cache the encoding the first time we query GetEncoding() and then always use that. We honor the environment encoding once and PSI can be used to update it, if the user wants.
3. Never cache the result, in which case we will always get the most current environment settings. PSI will still be there to override it but might be different from Console encoding settings done via managed layer.
4. Use 3 and make encoding setters on Console throw PlatformNotSupportedException or not expose Encodings on Console at all. This makes Process consistent but doing these just for Process is incorrect.
5. Take dependency on Console contract in Process (Conceptually, there is no layering break if we do this).

Among the options I would say we should use options 1 or 5. 2/3 are good solutions until Console.Encoding is exposed but that is only temporary. Thoughts?